src/account/utils/signature.cairo

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts for Cairo vX.Y.Z (account/utils/signature.cairo)
+
+use ecdsa::check_ecdsa_signature;
+use openzeppelin::account::eth_account::interface::EthPublicKey;
+use openzeppelin::account::utils::secp256k1::Secp256k1PointPartialEq;
+use starknet::eth_signature::{Signature, is_eth_signature_valid};
+use starknet::secp256_trait::{Secp256PointTrait, is_signature_entry_valid, recover_public_key};
+use starknet::secp256k1::Secp256k1Point;
+
+
+fn is_valid_signature(msg_hash: felt252, public_key: felt252, signature: Span<felt252>) -> bool {
+    let valid_length = signature.len() == 2;
+
+    if valid_length {
+        check_ecdsa_signature(msg_hash, public_key, *signature.at(0_u32), *signature.at(1_u32))
+    } else {
+        false
+    }
+}
+
+fn is_valid_eth_signature(
+    msg_hash: felt252, public_key: EthPublicKey, signature: Span<felt252>
+) -> bool {
+    let mut signature = signature;
```

Comment:
Because this function by design should not modify the given signature, mut is a requirement for deserialization, but should not modify the original signature passed down.